torchvision/models/vision_transformer.py

```diff
@@ -148,12 +159,32 @@ def __init__(
         self.num_classes = num_classes
         self.representation_size = representation_size
         self.norm_layer = norm_layer
+        self.conv_stem_layers = conv_stem_layers
 
         input_channels = 3
 
-        # The conv_proj is a more efficient version of reshaping, permuting
-        # and projecting the input
-        self.conv_proj = nn.Conv2d(input_channels, hidden_dim, kernel_size=patch_size, stride=patch_size)
+        if self.conv_stem_layers is not None:
+            # As per https://arxiv.org/abs/2106.14881
+            self.conv_proj = nn.Sequential()
+            prev_channels = input_channels
+            for i, conv_stem_layer in enumerate(self.conv_stem_layers):
+                if isinstance(conv_stem_layer, Dict):
+                    conv_stem_layer_params = ConvStemLayer(**conv_stem_layer)._asdict()
+                elif isinstance(conv_stem_layer, ConvStemLayer):
+                    conv_stem_layer_params = conv_stem_layer._asdict()
```

Comment:
The CI is [complaining](https://app.circleci.com/pipelines/github/pytorch/vision/14096/workflows/c10e5394-a469-46f8-b99a-6dda938e974a/jobs/1134575) about this. It would be best to avoid calling private methods and instead explicitly access the parameters that we want. Using the `ConvNormActivation` will help you do this easier anyway as you will only have to pass the parameters once.